models/data-service-postgresql.js

```diff
@@ -34,20 +34,25 @@ async function connect() {
  * Write data to Database
  *
  * @param {object} data - data object to be written
+ * @param {string} tableName - name of table to make query on
  *
  * @returns {object} write query result
  */
-async function write(data) {
+async function write(data, tableName) {
   try {
     const clientConnection = await connect();
     // Postgre SQL table columns
-    const { userEmail, jsonSavedData, visitedPages } = data;
-    // Postgre SQL table
-    const { tableName } = config.postgresql.additionalDatabaseConfig;
-
+    const columns = Object.keys(data);
+    const getPlaceholders = (input) => {
+      const placeholders = input.reduce((accumalator, currentValue, index) => {
+        accumalator.push(`$${index + 1}`);
+      });
+      return placeholders;
+    };
+    const values = Object.values(data);
     const queryString = {
-      text: `INSERT INTO ${tableName}(user_email, json_saved_data, visited_pages) VALUES ($1, $2, $3) RETURNING *`,
-      values: [userEmail, jsonSavedData, visitedPages],
+      text: `INSERT INTO ${tableName}(${columns}) VALUES (${getPlaceholders(columns)}) RETURNING *`,
+      values: values,
```

Comment:
Because it's the future, you can do:
```suggestion
      values,
```

(I probably wouldn't, though)